python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeCheckerBuilderTest.java

```diff
@@ -38,4 +40,68 @@ void typeSourceTest() {
       .isEqualTo(TriBool.FALSE);
   }
 
+  @Test
+  void isInstanceOfTest() {
+    var symbolTable = ProjectLevelSymbolTable.empty();
+    var table = new ProjectLevelTypeTable(symbolTable, new TypeShed(symbolTable));
+    var builder = new TypeCheckBuilder(table).isInstanceOf("int");
+
+    var intClassType = table.getType("int");
+    var strClassType = table.getType("str");
+
+    var aClassType = new ClassTypeBuilder()
+      .withName("A")
+      .withSuperClasses(intClassType)
+      .build();
+
+    var bClassType = new ClassTypeBuilder()
+      .withName("B")
+      .withSuperClasses(strClassType)
+      .build();
+
+    var cClassType = new ClassTypeBuilder()
+      .withName("C")
+      .withSuperClasses(PythonType.UNKNOWN, intClassType)
+      .build();
+
+    var dClassType = new ClassTypeBuilder()
+      .withName("D")
+      .withSuperClasses(PythonType.UNKNOWN, strClassType)
+      .build();
+
+    var eClassType = new ClassTypeBuilder()
+      .withName("E")
+      .withSuperClasses(aClassType, intClassType)
+      .build();
+
+    var intObjectType = new ObjectType(intClassType);
+    var strObjectType = new ObjectType(strClassType);
+    var aObject = new ObjectType(aClassType);
+    var bObject = new ObjectType(bClassType);
+    var cObject = new ObjectType(cClassType);
+    var dObject = new ObjectType(dClassType);
+    var eObject = new ObjectType(UnionType.or(intClassType, strClassType));
+    var fObject = UnionType.or(new ObjectType(intClassType), new ObjectType(strClassType));
+    var gObject = new ObjectType(UnionType.or(intClassType, aClassType));
+    var hObject = UnionType.or(new ObjectType(intClassType), new ObjectType(aClassType));
+    var iObject = new ObjectType(eClassType);
```

Comment:
It is confusing to have `iObject` of type `eClassType` while `eObject` is an object of another type. It would be good to make it consistent.